kivy/_event.pyx

```diff
@@ -894,12 +891,16 @@ cdef class EventObservers:
         cdef BoundCallback new_callback
 
         while callback is not None:
+            if is_ref and callback.is_ref:
+                f = callback.func()
+            else:
+                f = callback.func
             if (callback.lock != deleted and callback.largs is None and
-                callback.kwargs is None and callback.func == observer):
+                callback.kwargs is None and f == observer):
```

Comment:
I'm trying to think what would happen here, since `observer` might be a ref, but it is not derefed before comparison.